From 7eda76625bfec81fe902ab160f5aba46c02a0cea Mon Sep 17 00:00:00 2001
From: Ashok Mudithanapalli <ashokm@nvidia.com>
Date: Fri, 13 Nov 2015 17:47:24 +0530
Subject: [PATCH] audioflinger: Fix direct output pause

AudioFlinger avoids putting HAL in standby if track is in
paused state assuming Awesomeplayer will handle it. But
it causes problem for non-offloaded direct outputs.
Add condition check to fix that.

Change-Id: Ied7fbf43651ec80e3555d65a56f477761efee5c7
---

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 71fc498..6969819 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -4875,7 +4875,7 @@
                            mTracks[mTracks.size() - 1]->mState == TrackBase::IDLE;
     }
 
-    return !mStandby && !(trackPaused || (mHwPaused && !trackStopped));
+    return !mStandby && !((mType == OFFLOAD && trackPaused) || (mHwPaused && !trackStopped));
 }
 
 // getTrackName_l() must be called with ThreadBase::mLock held
